### DISCUSSION TITLE
Composer update with current developer version

### DISCUSSION THREAD

**Discussion by HLeithner** [clue/framework-x#67] [Q&A] [answered]
I'm not sure if my composer skills are to bad or I did something wrong but I'm failing to update my project composer dependencies withe the current framework-x commit.

Framework-X uses a special (upcoming) version of `react/event-loop`  by referencing a special commit as version 1.2.0.
`"react/event-loop": "dev-master#1a709e2 as 1.2.0",`
this works fine for framework-x it self but if I include framework-x in my application I get a dependency conflict.

```
Loading composer repositories with package information
Updating dependencies
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - clue/framework-x dev-main requires react/http ^1.1 -> satisfiable by react/http[v1.1.0, v1.2.0, v1.3.0].
    - react/http[v1.1.0, ..., v1.3.0] require react/event-loop ^1.0 || ^0.5 -> satisfiable by react/event-loop[v0.5.0, v0.5.1, v0.5.2, v0.5.3, v1.0.0, v1.1.0, v1.1.1].
    - You can only install one version of a package, so only one of these can be installed: react/event-loop[dev-master, v0.3.0, ..., v0.5.3, v1.0.0, v1.1.0, v1.1.1].
    - clue/framework-x dev-main requires react/event-loop dev-master#1a709e2 as 1.2.0 -> satisfiable by react/event-loop[dev-master].
    - Root composer.json requires clue/framework-x dev-main -> satisfiable by clue/framework-x[dev-main].

Use the option --with-all-dependencies (-W) to allow upgrades, downgrades and removals for packages currently locked to specific versions.
```

This can be fixed by adding `"react/event-loop": "dev-master#1a709e2 as 1.2.0",` to my own composer.json but I wonder if this is the right way?

**Comment by clue** [comment, marked as answer]
This is indeed a known issue that is caused by relying on an unreleased beta feature as discussed in https://github.com/clue/framework-x/issues/6.

We're working on resolving this as soon as possible. In the meantime, this can be worked around by installing like this:

```bash
$ composer require "clue/framework-x:dev-main" "react/event-loop:dev-master#1a709e2 as 1.2.0"
```

**Reply by HLeithner** [reply]
checking other issues first would prevent already answered questions ... thanks for the help

**Comment by clue** [comment]
This has now been fixed as discussed in #6. Installation should now work with the normal install instructions :+1:

**Reply by clue** [reply]
Just noticed I'm replying to @HLeithner who filed the upstream PR to address this installation issue 😃 Thank you!
